.github/workflows/gh-pages.yaml

```diff
@@ -42,7 +42,8 @@ jobs:
         make html
 
     - name: Deploy to GitHub Pages
+      if: github.event_name == 'push' && github.repository == 'microsoft/mattersim' && ( startsWith( github.ref, 'refs/tags/' ) || github.ref == 'refs/heads/main' )
```

Comment:
Thanks,

I purposely enable this for PR to main at the begining, but I agree with you that at this moment we probably only want the github pages to be built when pushed to main. 

However, why don't you just delete the following lines? Is the line you added different from doing so?

https://github.com/microsoft/mattersim/blob/f3acfbc740b85f1cf0d1f23067d7e22bf21f8749/.github/workflows/gh-pages.yaml#L8-L11

Reply:
If we delete the pull_request on main lines, we will only see failing doc builds when already merged to main. With my proposed change only the deployment is skipped in a pull request, while all other build steps are done and failures can be caught in a pull request before merging to main.